### PR TITLE
Fix formatNumber unit test

### PR DESCRIPTION
**Pull request by haacked**
When I ran this test on my machine (Mac OS) it failed.

Screen Shot 2020-05-16 at 12 35 16 PM

Even though the _Expected_ and _Actual_ values _look_ the same, they are not. If you look at the third character (index 2), the expected has an int value of `160` vs the actual of `8239`.

Screen Shot 2020-05-16 at 12 59 04 PM
Screen Shot 2020-05-16 at 12 59 18 PM

Digging into the code, this ultimately boils down to the `NumberFormatInfo.CurrencyGroupSeparator` value for `fr-FR`. On my machine, that value is `8239`.

I verified this by running [this code](https://dotnetfiddle.net/bCztS9) on https://dotnetfiddle.net/.

```csharp
using System;
using System.Globalization;
					
public class Program
{
	public static void Main()
	{
		var separator = CultureInfo.GetCultureInfo("fr-FR")
			.NumberFormat
			.CurrencyGroupSeparator;
		
		Console.WriteLine(
			$"Currency Group Separator for fr-FR is {(int)separator[0]}");
		Console.WriteLine(
			$"As HEX {((int)separator[0]).ToString("X")}");
	}
}
```

That outputs

> Currency Group Separator for fr-FR is 8239
> As HEX 202F

This PR fixes the test by correcting the expected value and now the test passes on my machine.

### CODE REVIEW

File: tests/AdaptiveExpressions.Tests/ExpressionParserTests.cs [hunk 1]
```diff
@@ -635,7 +635,7 @@ public class ExpressionParserTests
             Test("formatNumber(12.123, 2)", "12.12"),
             Test("formatNumber(1.551, 2)", "1.55"),
             Test("formatNumber(12.123, 4)", "12.1230"),
-            Test("formatNumber(12000.3, 4, 'fr-fr')", "12\x00a0000,3000"),
+            Test("formatNumber(12000.3, 4, 'fr-fr')", "12\x202F000,3000"),
             #endregion
```
Comment (chrimc62) [inline tests/AdaptiveExpressions.Tests/ExpressionParserTests.cs:639]: There is something odd here.  Your fix fails on our build machines when they run the Mac/Linux profile.  You can see this in the build logs from your PR page. The question is why your results are different.  I'm not that familiar with the Mac, but is there the equivalent of Windows language packs?  Is it possible that it depends on whether or not you have that installed?
 X Evaluate (formatNumber(12000.3, 4, 'fr-fr'),12â€¯000,3000,) [27ms]
2020-05-16T20:55:28.5352450Z   Error Message:
2020-05-16T20:55:28.5353480Z    Assert.AreEqual failed. Expected:<12â€¯000,3000>. Actual:<12Â 000,3000>. 
2020-05-16T20:55:28.5354020Z   Stack Trace:
2020-05-16T20:55:28.5354700Z      at AdaptiveExpressions.Tests.ExpressionParserTests.AssertObjectEquals(Object expected, Object actual) in /Users/runner/runners/2.168.2/work/1/s/tests/AdaptiveExpressions.Tests/ExpressionParserTests.cs:line 1181
2020-05-16T20:55:28.5355800Z    at AdaptiveExpressions.Tests.ExpressionParserTests.Evaluate(String input, Object expected, HashSet`1 expectedRefs) in /Users/runner/runners/2.168.2/work/1/s/tests/AdaptiveExpressions.Tests/ExpressionParserTests.cs:line 964

**Comment by haacked** [comment]
Thanks for looking at this over the weekend. You're going to love what I discovered. 😛 

I installed the French language on my mac and still got the same results. I even made it my only and primary language and restarted my machine. Still the same results. Then I dug into it and learned something interesting.

According to the Unicode Common Locale Data Repository, the Unicode Consortium changed the thousands separator for french from `NBSP` (aka `U+00A0) to `NNBSP` (aka `U+202F`).

You can see this in the [CLDR 34 release notes](http://cldr.unicode.org/index/downloads/cldr-34).

> The French locale now uses narrow no-break space U+202F is several places: as the numeric grouping separator, in many short unit patterns, and in the locale display name patterns. It also changed normal space to no-break space U+00A0 in the wide unit patterns.

So it seems that the behavior on my Mac and the [behavior on dotnetfiddle](https://dotnetfiddle.net/YDLsGS) is correct.

BTW, I realized I made a small mistake in my Fiddle (corrected in the new link). When setting a culture for unit tests, it's important to use the constructor so you can override the default behavior and ignore user overrides on the local system. Not sure if that's affecting your tests or not.

```csharp
var separator = new CultureInfo("fr-FR", false)
			.NumberFormat
			.CurrencyGroupSeparator;
```

**Comment by cleemullins** [comment]
@haacked This is a fascintaing bug. That the codepoint changed is just... well, I've never seen hat before. 

Giving this to our resident Unicode Expert, @johnataylor

**Comment by cleemullins** [comment]
@johnataylor  John, could you look at this? This is one of the stranged bugs I've seen, and I think the resolution is to have the Unit Test treat both answers as correct.

**Comment by cleemullins** [comment]
@johnataylor Could you review this and let us know your thinking. Would like to merge ASAP.

**Comment by cleemullins** [comment]
@haacked Thanks Phil - the level of Unicode confusion here (to me) is crazy. I've not seen anything that weird in ages. Thanks from pushing this in!!

**Comment by haacked** [comment]
@cleemullins my pleasure! Such an interesting mystery. 😄

**Pull request merged**